exam/src/main/java/in/testpress/exam/ui/TestActivity.java

```diff
@@ -579,12 +580,40 @@ private void displayTestFragment(Bundle arguments) {
         if (getSupportActionBar() != null) {
             getSupportActionBar().hide();
         }
+
+        if (examInstructionExists()){
+            displayExamInstruction(arguments);
+            return;
+        }
+
+        startExam(arguments);
+    }
+
+    private void displayExamInstruction(Bundle arguments){
+        TestInstructions instructions = new TestInstructions(() -> {
+            startExam(arguments);
+            return null;
+        });
+
+        Bundle instructionArguments = new Bundle();
+        instructionArguments.putString(TestInstructions.Companion.getINSTRUCTIONFLAG(), exam.getInstructions());
+        instructions.setArguments(instructionArguments);
```

Comment:
Wildcard arguments should be avoided. Pass only required arguments

Reply:
These can be moved to that new fragment itself. You can create a companion method called `createInstance`